lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -486,9 +491,6 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                     if node.name != self.unit_name
                 ]
                 self._compute_and_broadcast_updated_topology(remaining_nodes)
-            elif self.app.planned_units() == 0:
```

Comment:
Removing this check as the unit is going away and it is the last one of its own peers. This logic will result in cleaning an empty peer data bag.

Reply:
I'm not sure that's accurate, the first hook to execute during the teardown flow is `storage_detaching` - we still have the peer relation data bag then. 
(It was proved to successfully work before, no?)

Reply:
@Mehdi-Bendriss you're right if you are going away because of a `remove-unit`. Then, at `storage-detaching`, you will have:
```
2024-08-09 13:34:54,371 DEBUG    ops 2.15.0 up and running.
2024-08-09 13:34:54,450 DEBUG    Emitting Juju event opensearch_data_storage_detaching.
(Pdb) p self.model.get_relation(PeerRelationName)
<ops.model.Relation opensearch-peers:1>
```

Whereas, if we do `remove-application`, then we get:
```
2024-08-09 13:50:33,992 DEBUG    ops 2.15.0 up and running.
2024-08-09 13:50:34,065 DEBUG    Emitting Juju event opensearch_data_storage_detaching.
> /var/lib/juju/agents/unit-opensearch-0/charm/lib/charms/opensearch/v0/opensearch_base_charm.py(479)_on_opensearch_data_storage_detaching()
-> if not self.node_lock.acquired:
(Pdb) p self.model.get_relation(PeerRelationName)
None
```

In this case, the lines below will give:
```
(Pdb) self.peers_data.delete(Scope.APP, "bootstrap_contributors_count")
*** KeyError: 'bootstrap_contributors_count'
```

Not sure what changed before and now. We did update `ops` to 2.15 and `juju` to 3.4.5. I am using juju 3.4.5, status: https://pastebin.ubuntu.com/p/YgVSbqWsGW/

Reply:
I think we can keep that check and add an extra check if the peer relation still exists.